sync-core/src/test/java/com/cloudant/sync/query/QueryExecutorMatcherTest.java

```diff
@@ -0,0 +1,39 @@
+//  Copyright (c) 2014 Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+
+package com.cloudant.sync.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import com.cloudant.sync.util.SQLDatabaseTestUtils;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class QueryExecutorMatcherTest extends AbstractQueryTestBase {
```

Comment:
Added comments where appropriate explaining what each class does.  I took a note to revisit the process later on in both iOS and Android once the smoke clears.
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-0eaf4604dd4e002edfe0d41fdea2c7e3R20
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-162b0c09f735c14ea05abac3dd234d9eR27
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-c4ac637a25d1bf100c90b6b658737b56R20
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-253a3500c1d06e06260c5b09832b8986R21
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-c8d8ef5f4948719512f62057f5359599R24
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-162fe9eab99c2b46b7150404a5da366dR24